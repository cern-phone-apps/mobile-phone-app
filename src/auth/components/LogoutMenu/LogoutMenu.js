import React from 'react';
import Icon from 'react-native-vector-icons/Ionicons';
import HeaderButtons, {
  HeaderButton,
  Item
} from 'react-navigation-header-buttons';
import PropTypes from 'prop-types';
import { Alert } from 'react-native';

const IoniconsHeaderButton = passMeFurther => (
  // the `passMeFurther` variable here contains props from <Item .../> as well as <HeaderButtons ... />
  // and it is important to pass those props to `HeaderButton`
  // then you may add some information like icon size or color (if you use icons)
  <HeaderButton
    {...passMeFurther}
    IconComponent={Icon}
    iconSize={25}
    color="white"
  />
);

const disconnectUserAction = async (logout, navigation, phoneService) => {
  console.log(`Disconnecting user`);
  phoneService.disconnectUser();
  logout();
  navigation.navigate('Auth');
};

const logoutFunc = (logout, navigation, phoneService) => {
  Alert.alert('Log out?', 'Are you sure you want to log out?', [
    {
      text: 'Cancel',
      style: 'cancel'
    },
    {
      text: 'Log Out',
      onPress: () => {
        disconnectUserAction(logout, navigation, phoneService);
      }
    }
  ]);
};

const LogoutMenu = ({ logout, navigation, phoneService }) => (
  <HeaderButtons
    OverflowIcon={<Icon name="md-more" size={23} color="white" />}
    HeaderButtonComponent={IoniconsHeaderButton}
  >
    <Item
      testID="LogoutMenu"
      title="Log out"
      onPress={() => logoutFunc(logout, navigation, phoneService)}
      show="never"
    />
  </HeaderButtons>
);

LogoutMenu.propTypes = {
  logout: PropTypes.func.isRequired,
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired
  }).isRequired,
  phoneService: PropTypes.shape({
    disconnectUser: PropTypes.func.isRequired
  }).isRequired
};

export default LogoutMenu;
