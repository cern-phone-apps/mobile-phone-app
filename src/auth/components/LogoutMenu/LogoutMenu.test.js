import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import LogoutMenu from './LogoutMenu';

const logout = jest.fn(() => {});

const phoneService = {
  disconnectUser: jest.fn(() => {})
};

const navigation = {
  navigate: jest.fn(() => {})
};

test('Render the component', async () => {
  const { baseElement } = render(
    <LogoutMenu
      logout={logout}
      phoneService={phoneService}
      navigation={navigation}
    />
  );
  expect(baseElement).toMatchSnapshot();
});
