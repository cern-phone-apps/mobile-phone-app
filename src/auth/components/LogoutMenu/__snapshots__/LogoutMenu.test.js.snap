// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render the component 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          Object {
            "marginRight": 5,
          },
        ]
      }
    >
      <View>
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "height": 1,
              "position": "absolute",
              "right": 0,
              "top": -10,
              "width": 1,
            }
          }
        />
        <TouchableOpacity
          activeOpacity={0.2}
          disabled={false}
          hitSlop={
            Object {
              "bottom": 5,
              "left": 5,
              "right": 5,
              "top": 5,
            }
          }
          style={
            Array [
              Object {
                "alignItems": "center",
                "justifyContent": "center",
              },
              Array [
                Object {
                  "marginTop": 2,
                },
                undefined,
              ],
            ]
          }
          testID="headerOverflowButton"
        >
          <View>
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 23,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Ionicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              ï‡‰
            </Text>
          </View>
        </TouchableOpacity>
      </View>
    </View>
  </View>
</View>
`;
